impeller/entity/shaders/filters/gaussian.frag

```diff
@@ -11,14 +11,11 @@ uniform f16sampler2D texture_sampler;
 
 layout(constant_id = 0) const float supports_decal = 1.0;
 
-struct KernelSample {
-  vec2 uv_offset;
-  float coefficient;
-};
-
-uniform KernelSamples {
-  int sample_count;
-  KernelSample samples[50];
+uniform BlurInfo {
+  float sample_count;
+
+  // X, Y are uv offset and Z is Coefficient. W is padding.
+  vec4 sample_data[50];
```

Comment:
If we jam this into a vec4 then we don't have to deal with removing padding elements while uploading (see buffer binding gles).

Reply:
Nice!